File: jaeseuk/pypython/Programmers/level1/recommend_a_job.py
def solution(table, languages, preference):
    table = [job.split() for job in table]
    answer = [0, ""]

    for job in table:
        score = 0
        for i in range(len(languages)):
            if languages[i] in job:
                score += (6 - job.index(languages[i])) * preference[i]

        answer = min(answer, [score, job[0]], key=lambda x: (-x[0], x[1]))

    return answer[1]

File: jaeseuk/kokotlin/src/main/kotlin/RecommendAJob.kt
class RecommendAJob {
    fun solution(table: Array<String>, languages: Array<String>, preference: IntArray): String {
        var answer = Pair<Int, String>(0, "")

        table.map { it.split(" ") }.forEach {
            var score = 0
            for (lan in languages zip preference.toTypedArray()) {
                if (lan.first in it) {
                    score += (6 - it.indexOf(lan.first)) * lan.second
                }
            }

            answer = if (answer.first > score) {
                answer
            } else if (answer.first == score) {
                if (answer.second > it[0]) Pair(score, it[0]) else answer
            } else {
                Pair(score, it[0])
            }
        }
        return answer.second
    }
}